cmd/importcmd/imagebundle/image_bundle.go

```diff
@@ -43,22 +44,42 @@ func NewCommand(out output.Output) *cobra.Command {
 			cleaner.AddCleanupFn(func() { _ = os.RemoveAll(tempDir) })
 			out.EndOperation(true)
 
-			out.StartOperation("Unarchiving image bundle")
-			err = archive.UnarchiveToDirectory(imageBundleFile, tempDir)
-			if err != nil {
-				out.EndOperation(false)
-				return fmt.Errorf("failed to unarchive image bundle: %w", err)
-			}
-			out.EndOperation(true)
+			sort.Strings(imageBundleFiles)
 
-			out.StartOperation("Parsing image bundle config")
-			cfg, err := config.ParseImagesConfigFile(filepath.Join(tempDir, "images.yaml"))
-			if err != nil {
-				out.EndOperation(false)
-				return err
+			var cfg config.ImagesConfig
+
+			// Just in case users specify the same bundle twice, keep a track of
+			// files that have been extracted already so we only extract each of them once.
+			extractedBundles := make(map[string]struct{}, len(imageBundleFiles))
+
+			for _, imageBundleFile := range imageBundleFiles {
+				if _, ok := extractedBundles[imageBundleFile]; ok {
+					continue
+				}
+				extractedBundles[imageBundleFile] = struct{}{}
+				out.StartOperation(fmt.Sprintf("Unarchiving image bundle %q", imageBundleFile))
+				err = archive.UnarchiveToDirectory(imageBundleFile, tempDir)
+				if err != nil {
+					out.EndOperation(false)
+					return fmt.Errorf("failed to unarchive image bundle: %w", err)
+				}
+				out.EndOperation(true)
+
+				out.StartOperation("Parsing image bundle config")
+				bundleCfg, err := config.ParseImagesConfigFile(
+					filepath.Join(tempDir, "images.yaml"),
```

Comment:
Took a little to understand how this works, but this `images.yaml` will always be overwritten and from the bundle in this loop, and then merged with the rest of the cfg.

Reply:
Ah yeah let me add a comment

Reply:
Added in 73e4c43003159de58b1f3b799fa76b31666510f8